storm-core/src/jvm/backtype/storm/utils/ShellProcess.java

```diff
@@ -121,34 +117,15 @@ public void writeTaskIds(List<Integer> taskIds) throws IOException {
     }
 
     public void logErrorStream() {
-        try {
-            while (processErrorStream.available() > 0) {
-                int bufferSize = processErrorStream.available();
-                byte[] errorReadingBuffer = new byte[bufferSize];
-                processErrorStream.read(errorReadingBuffer, 0, bufferSize);
-                ShellLogger.info(new String(errorReadingBuffer));
-            }
-        } catch (Exception e) {
+        final Optional<String> errorsString = getErrorStreamMessage();
+        if (errorsString.isPresent()) {
+            ShellLogger.info(errorsString.get());
         }
     }
 
     public String getErrorsString() {
```

Comment:
add a comment saying this method must never block since it is run during critical shutdown phase